analyzers/tests/SonarAnalyzer.UnitTest/Rules/DoNotCatchSystemExceptionTest.cs

```diff
@@ -25,18 +25,26 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class DoNotCatchSystemExceptionTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<DoNotCatchSystemException>();
+
         [TestMethod]
         public void DoNotCatchSystemException() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\DoNotCatchSystemException.cs", new DoNotCatchSystemException());
+            builder.AddPaths("DoNotCatchSystemException.cs").Verify();
+
+        [TestMethod]
+        public void DoNotCatchSystemException_AzureFunction() => builder
+            .AddReferences(NuGetMetadataReference.MicrosoftAzureWebJobsCore())
+            .AddPaths("DoNotCatchSystemException.AzureFunction.cs")
+            .Verify();
```

Comment:
> As long as AzureFunction means only that function and not the whole project

That's my concern. If you look at it, you don't even need to have the `[FunctionName]` attribute to be in an AzureFunction like environment. It is enough to have *Microsoft.Azure.WebJobs.Core* referenced. Take for instance a look at this https://docs.microsoft.com/en-us/azure/azure-functions/functions-compare-logic-apps-ms-flow-webjobs#compare-functions-and-webjobs

There is a technology called "Azure App Service WebJobs". It is using the same SDK and supports (almost) the same triggers. The only difference is, that you have access to the WebServer that hosts the Function (it is also free of charge if you already use the Azure App Service to host a web application. That's why we used it in our last project instead of the  AzureFunction service).

These "Functions" are not bound via `[FunctionName]` but you have to do the binding yourself (see e.g. https://docs.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started#add-a-function).

Chances are, that we will change `AzureFunctionMethod()` in a way that we check for the presence of some type of *Microsoft.Azure.WebJobs.Core* in the compilation. But if we do that, it would even more useful to contribute the analyzer to the WebJobs SDK to live there.

Reply:
This rule honestly doesn't care. It's the S6421 that should ask and allow catching all.

Regardless of that, with the current implementation, we don't need to separate it to test the currently intended functionality of this rule.